### PR TITLE
samples: added new translate document samples

### PR DESCRIPTION
**Pull request by munkhuushmgl**
Googlers, see internal issue 187405485.

canonical samples:
https://github.com/googleapis/python-translate/pull/183/files

timeout issue has been tracked in the following gh issue:
#613

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+      GcsDestination gcsDestination =
+          GcsDestination.newBuilder().setOutputUriPrefix(outputUri).build();
+
+      BatchDocumentOutputConfig batchDocumentOutputConfig =
+          BatchDocumentOutputConfig.newBuilder().setGcsDestination(gcsDestination).build();
+
+      BatchTranslateDocumentRequest request =
+          BatchTranslateDocumentRequest.newBuilder()
+              .setParent(parent.toString())
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java:103]: why doesn't this method take a LocationName?

weird that parent has two types

File: samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  public static void translateDocument() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "YOUR-PROJECT-ID";
+    String filePath = "your-text";
+    translateDocument(projectId, filePath);
+  }
+
+  // Translating Document
+  public static void translateDocument(String projectId, String filePath) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java:41]: same as above

File: samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+              .setDocumentInputConfig(documentInputConfig)
+              .build();
+
+      TranslateDocumentResponse response = client.translateDocument(request);
+
+      // To view translated document, write `response.document_translation.byte_stream_outputs`
+      // to file. If not provided in the TranslationRequest, the translated file will only be
+      // returned through a byte-stream and its output mime type will be the same as the input
+      // file's mime type
+      System.out.format(
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java:70]: just use string concatenation. It's simpler and easier to read

File: samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java (hunk 1 above)
Comment (munkhuushmgl) [reply samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java:103]: Thats what I thought too.

**Review by telpirion — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      OperationFuture<BatchTranslateDocumentResponse, BatchTranslateDocumentMetadata> future =
+          client.batchTranslateDocumentAsync(request);
+
+      System.out.println("Waiting for operation to complete...");
+
+      // random number between timeout
+      long randomNumber = ThreadLocalRandom.current().nextInt(timeout, timeout + 100);
+      BatchTranslateDocumentResponse response = future.get(randomNumber, TimeUnit.SECONDS);
+
+      System.out.printf("Total Pages: %d\n", response.getTotalPages());
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java:119]: system.out.println with string concatenation is clearer and simpler

File: samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "YOUR-PROJECT-ID";
+    String filePath = "your-text";
+    translateDocument(projectId, filePath);
+  }
+
+  // Translating Document
+  public static void translateDocument(String projectId, String filePath) throws IOException {
+    // Initialize client that sends requests. This client can be reused for multiple requests. After
+    // completing all of your requests, call the "close" method on the client to delete safely clean
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/translatev3beta1/TranslateDocument.java:42]: delete "delete safely"

File: samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java (hunk 1 above)
Comment (telpirion) [reply samples/snippets/src/main/java/com/example/translatev3beta1/BatchTranslateDocument.java:103]: It looks like the [public protos are missing the `google.api.resource_reference` annotation](https://github.com/googleapis/googleapis/blob/master/google/cloud/translate/v3beta1/translation_service.proto#L1197). I have filed an internal issue for this--Googlers, see issue #196978761

**Comment by telpirion** [comment]
Blocked by #774

**Review by nicain — APPROVED** [review]

**Review by aribray — APPROVED** [review]
Merging because we've identified the next steps for closing out the related issue.

**Review by eaball35 — APPROVED** [review]

**Pull request merged**